compiler/src/dotty/tools/dotc/typer/CrossVersionChecks.scala

```diff
@@ -107,6 +109,18 @@ class CrossVersionChecks extends MiniPhase:
     }
   }
 
+  /** ??? */
+  def checkDeprecatedInheritance(parents: List[Tree])(using Context): Unit = {
+    for parent <- parents
+        psym = parent.tpe.classSymbol
+        if !skipWarning(psym)
+        annot <- psym.getAnnotation(defn.DeprecatedInheritanceAnnot)
```

Comment:
It's cheaper to check the annotation than to walk owners checking whether to skipWarning, also the annotation is rare, so it's more natural to reverse the lines. It's a matter of style (?) whether to have trailing if in the for comprehension or move it to the body.

Reply:
I've actually put the if after but I agree, I will switch them